test/py/CMakeLists.txt

```diff
@@ -23,6 +23,23 @@
 #####################################################################################
 
 include(PythonModules)
+set(FIX_NAME SETUP_VIRTUAL_ENV)
+set(VENV_DIR ${CMAKE_BINARY_DIR}/test/py/venv)
+
+function(add_py_venv_fixture FIXTURE_NAME VIRTUAL_ENV_DIR REQUIREMENTS_FILE)
+    message("Attempting to create virtual environment: ${VIRTUAL_ENV_DIR}")
+    find_package (Python3 COMPONENTS Interpreter)
```

Comment:
This should use the `foreach(PYTHON_VERSION ${PYTHON_VERSIONS})` to add fixtures for all python version and then it needs use `${PYTHON_${PYTHON_VERSION}_EXECUTABLE}` to get the executable. Dont call `find_package(Python3)` as this could get a different python version.